packages/cadl-ranch-specs/http/payload/xml/mockapi.ts

```diff
@@ -0,0 +1,117 @@
+import { passOnSuccess, mockapi, xml } from "@azure-tools/cadl-ranch-api";
+import { ScenarioMockApi } from "@azure-tools/cadl-ranch-api";
+
+export const Scenarios: Record<string, ScenarioMockApi> = {};
+
+const simpleModel = `
+<SimpleModel>
+  <Name>foo</Name>
+  <Age>123</Age>
+</SimpleModel>
+`;
+
+const modelWithSimpleArrays = `
+<ModelWithSimpleArrays>
+  <Colors>red</Colors>
+  <Colors>green</Colors>
+  <Colors>blue</Colors>
+  <Counts>1</Counts>
+  <Counts>2</Counts>
+</ModelWithSimpleArrays>
+`;
+
+const modelWithArrayOfModel = `
+<ModelWithArrayOfModel>
+  <SimpleModel>
+    <Name>foo</Name>
+    <Age>123</Age>
+  </SimpleModel>
+  <SimpleModel>
+    <Name>bar</Name>
+    <Age>456</Age>
+  </SimpleModel>
+</ModelWithArrayOfModel>
+`;
+
+const modelWithOptionalField = `
+<ModelWithOptionalField>
+  <Item>widget</Item>
+  <Counts />
```

Comment:
For optional fields, should we use an empty element (i.e. _not_ omit it)?

Reply:
I think you wouldn't pass the field at all

Reply:
Do you think omitting it is the correct default behavior? I bring this up because we had a bug in the Go SDK where tables would sometimes send an empty element (a timestamp IIRC) and we didn't properly handle that case. If an empty element is the uncommon case for "no value" then I can add a separate test for that.

Reply:
if I see this I kinda see it equivalent to sending that as json
```json
{item: "widget", counts: []}
```
instead of 
```json
{item:  "widget"}
```

Reply:
Here, `counts` is an `int32`, not array, but I see your point (I'm going to rename this field to avoid confusion with other models).

This feels similar to JSON nulls. For GETs, it seems that omission of the `counts` field or an empty element are semantically equivalent. I've seen this in the wild, so we do need to ensure that generated code properly handles it.

But for PUTs etc, omission vs empty element is _not_ equivalent. If I omit the element, then it means "preserve the existing value". But if I send an empty element, then it means "erase the existing value". I don't know if this is a real scenario though.

Reply:
Checked the blob storage [REST API docs](https://learn.microsoft.com/rest/api/storageservices/set-blob-service-properties?tabs=microsoft-entra-id) and the omission vs empty element on PUT is real.

I will change this test to cover the omission scenario and add separate test for empty element.

Reply:
yeah exactly, for serializing I assume you wouldn't put it ever but for deserializing it I guess it should be equivalent as long as you have the schema that says what type it expect